checkoutmanager/config.py

```diff
@@ -116,6 +116,12 @@ def directories(self, group=None):
                 result.append(dirinfoclass(directory, url))
         return sorted(result)
 
+    def directory_from_url(self, url):
+        return [x for x in self.directories() if x.url == url][0]
```

Comment:
I looked at the travis test results and it is indeed an IndexError that's the problem there.